code/game/gamemodes/cult/blood_magic.dm

```diff
@@ -0,0 +1,662 @@
+		return
+	if(iscultist(user))
+		for(var/R in teleport_runes)
+			var/obj/effect/rune/teleport/T = R
+			var/resultkey = T.listkey
+			if(resultkey in teleportnames)
+				duplicaterunecount[resultkey]++
+				resultkey = "[resultkey] ([duplicaterunecount[resultkey]])"
+			else
+				teleportnames.Add(resultkey)
+				duplicaterunecount[resultkey] = 1
+			potential_runes[resultkey] = T
+
+		if(!potential_runes.len)
+			to_chat(user, "<span class='warning'>There are no valid runes to teleport to!</span>")
+			log_game("Teleport spell failed - no other teleport runes")
+			return ..(user, 0)
+
+		if(!is_level_reachable(user.z))
+			to_chat(user, "<span class='cultitalic'>You are not in the right dimension!</span>")
+			log_game("Teleport spell failed - user in away mission")
+			return ..(user, 0)
+		uses--
+		var/mob/living/L = target
+		var/input_rune_key = input(user, "Choose a rune to teleport to.", "Rune to Teleport to") as null|anything in potential_runes //we know what key they picked
+		var/obj/effect/rune/teleport/actual_selected_rune = potential_runes[input_rune_key] //what rune does that key correspond to?
+		if(!src || QDELETED(src) || !user || user.l_hand != src && user.r_hand != src || user.incapacitated() || !actual_selected_rune)
+			return ..(user, 0)
+		var/turf/origin = get_turf(user)
+		var/turf/destination = get_turf(actual_selected_rune)
+		new /obj/effect/temp_visual/dir_setting/cult/phase/out(origin)
+		new /obj/effect/temp_visual/dir_setting/cult/phase(destination)
+		if(is_mining_level(user.z) && !is_mining_level(destination.z)) //No effect if you stay on lavaland
+			actual_selected_rune.handle_portal("lava")
+		else if(!is_station_level(user.z) || istype(get_area(user), /area/space))
+			actual_selected_rune.handle_portal("space", origin)
+		if(user == target)
+			target.visible_message("<span class='warning'>Dust flows from [user]'s hand, and [user.p_they()] disappear[user.p_s()] in a flash of red light!</span>", \
+							"<span class='cultitalic'>You speak the words and find yourself somewhere else!</span>")
+		else
+			target.visible_message("<span class='warning'>Dust flows from [user]'s hand, and [target] disappears in a flash of red light!</span>", \
+						"<span class='cultitalic'>You suddenly find yourself somewhere else!</span>")
+		destination.visible_message("<span class='warning'>There is a boom of outrushing air as something appears above the rune!</span>", null, "<i>You hear a boom.</i>")
+		L.forceMove(destination)
+		return ..()
+
+//Shackles
+/obj/item/melee/blood_magic/shackles
+	name = "Shackling Aura"
+	desc = "Will start handcuffing a victim on contact, and mute them if successful."
+	invocation = "In'totum Lig'abis!"
+	color = "#000000" // black
+
+/obj/item/melee/blood_magic/shackles/afterattack(atom/target, mob/living/carbon/user, proximity)
+	if(iscultist(user) && iscarbon(target) && proximity)
+		var/mob/living/carbon/C = target
+		if(C.get_num_arms(FALSE) >= 2 || C.get_arm_ignore())
```

Comment:
Can use canBeHandcuffed for this. Although that proc is not used by normal cuffs... it's a bit of a mess